check_domain_expiration.sh

```diff
@@ -257,29 +255,106 @@ check_domain()
 				EXDATE=`date -d"$EXDATE_TMP" +%Y-%m-%d`
 				EXP_DAYS=$(( ( $(date -ud ${EXDATE} +'%s') - $(date -ud `date +%Y-%m-%d` +'%s') )/60/60/24 ))
 		fi
+
+
+	elif [ "$DTYPE" == "tech" ]
+	then
+		EXDATE_TMP=$(${WHOIS} -h whois.nic.tech "${1}" | grep -i 'Expiry Date' | ${AWK} '{ print $4 }' ) 
+		if [ -z "$EXDATE_TMP" ]
+			then
+				EXP_DAYS=NULL
+			else
+				EXDATE=`date -d"$EXDATE_TMP" +%Y-%m-%d`
+				EXP_DAYS=$(( ( $(date -ud ${EXDATE} +'%s') - $(date -ud `date +%Y-%m-%d` +'%s') )/60/60/24 ))
+		fi
+
+
+	elif [ "$DTYPE" == "co" ]
+	then
+		EXDATE_TMP=$(${WHOIS} -h whois.nic.co "${1}" | grep -i 'Expiry Date' | ${AWK} '{ print $4 }' ) 
+		if [ -z "$EXDATE_TMP" ]
+			then
+				EXP_DAYS=NULL
+			else
+				EXDATE=`date -d"$EXDATE_TMP" +%Y-%m-%d`
+				EXP_DAYS=$(( ( $(date -ud ${EXDATE} +'%s') - $(date -ud `date +%Y-%m-%d` +'%s') )/60/60/24 ))
+		fi
+
+
+	elif [ "$DTYPE" == "digital" ]
+	then
+		EXDATE_TMP=$(${WHOIS} -h whois.nic.digital "${1}" | grep -i 'Expiry Date' | ${AWK} '{ print $4 }' ) 
+		if [ -z "$EXDATE_TMP" ]
+			then
+				EXP_DAYS=NULL
+			else
+				EXDATE=`date -d"$EXDATE_TMP" +%Y-%m-%d`
+				EXP_DAYS=$(( ( $(date -ud ${EXDATE} +'%s') - $(date -ud `date +%Y-%m-%d` +'%s') )/60/60/24 ))
+		fi
+
+
 	else
 		echo "UNKNOWN - "$DTYPE" unsupported"
 		exit 3
 	fi
 }
 
+# Check function for a specific whois
+check_domain_by_whois()
+{
+	# Set domain
+	DOMAIN=$1
+	SERVER=$2
+
+	if [ "$SERVER" == "whois.crazydomains.com" ]
+	then
+		EXDATE_TMP=$(${WHOIS} -h ${SERVER} "${1}" | grep -i 'Expiration Date' | ${AWK} '{ print $5 }' ) 
+		if [ -z "$EXDATE_TMP" ]
+		then
+			EXP_DAYS=NULL
+		else
+			EXDATE=`date -d"$EXDATE_TMP" +%Y-%m-%d`
+			EXP_DAYS=$(( ( $(date -ud ${EXDATE} +'%s') - $(date -ud `date +%Y-%m-%d` +'%s') )/60/60/24 ))
+		fi
+
+
+	elif [ "$SERVER" == "whois.cloudflare.com" ]
+	then
+		EXDATE_TMP=$(${WHOIS} -h ${SERVER} "${1}" | grep -i 'Expiration Date' | ${AWK} '{ print $5 }' ) 
```

Comment:
You've set `DOMAIN` variable at the beginning of this function, so maybe it would be better to use it instead of `${1}`? As now `DOMAIN` is not used in the entire function.

Reply:
Done